### DISCUSSION TITLE
After many sip calls got stream: failed to create socket for media 'audio' (Too many open files)

### DISCUSSION THREAD

**Discussion by assegaf** [baresip/baresip#2052] [Q&A] [answered]
Hi,

After so many calls both inbound and outbound, I got error when receiving calls (replied with 500 internal error), this log said it all about too many file open. Unfortunately its run on android, so I cant debug more and find out what file open causing this.

BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:31.359  4203  4694 D BSLib   : main: fd
08-03 15:10:31.359  4203  4694 D BSLib   : _listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:31.359  4203  4694 D BSLib   : stream: rtp_listen failed: af=AF_INET ports=1024-49152 (Too many open files
08-03 15:10:31.360  4203  4694 D BSLib   : )
08-03 15:10:31.360  4203  4694 D BSLib   : stream: failed to create socket for media 'audio' (Too many open files)
08-03 15:10:31.360  4203  4694 D BSLib   : audio: destroyed (started=0)
08-03 15:10:31.360  4203  4694 D BSLib   : ua: ca
08-03 15:10:31.360  4203  4694 D BSLib   : ll_alloc: Too many open files
08-03 15:10:35.132  4203  4694 D BSLib   : ua: sipsess connect via UDP 192.168.20.200:5060 --> 192.168.20.131:49339
08-03 15:10:35.132  4203  4694 D BSLib   : ua: using AF from sdp offer: af=AF_INET
08-03 15:10:35.132  4203  4694 D BSLib   : call: allo
08-03 15:10:35.132  4203  4694 D BSLib   : c with params laddr=192.168.20.131, af=AF_INET, use_rtp=1
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main
08-03 15:10:35.132  4203  4694 D BSLib   : : fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_li
08-03 15:10:35.132  4203  4694 D BSLib   : sten: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: f
08-03 15:10:35.132  4203  4694 D BSLib   : d=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 
08-03 15:10:35.132  4203  4694 D BSLib   : flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.132  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0
08-03 15:10:35.133  4203  4694 D BSLib   : x01 - Max 1024 fds
08-03 15:10:35.133  4203  4694 D BSLib   : 
08-03 15:10:35.133  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.133  4203  4694 D BSLib   : 1204 flags=0x
08-03 15:10:35.133  4203  4694 D BSLib   : 01 - Max 1024 fds
08-03 15:10:35.133  4203  4694 D BSLib   : 
08-03 15:10:35.136  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.136  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.136  4203  4694 D BSLib   : 
08-03 15:10:35.137  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.137  4203  4694 D BSLib   : 
08-03 15:10:35.137  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.137  4203  4694 D BSLib   : 
08-03 15:10:35.137  4203  4694 D BSLib   : 
08-03 15:10:35.138  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.138  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.138  4203  4694 D BSLib   : main: f
08-03 15:10:35.138  4203  4694 D BSLib   : d_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.138  4203  4694 D BSLib   : 
08-03 15:10:35.138  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.138  4203  4694 D BSLib   : 
08-03 15:10:35.139  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x
08-03 15:10:35.139  4203  4694 D BSLib   : 01 - Max 1024 fds
08-03 15:10:35.139  4203  4694 D BSLib   : 
08-03 15:10:35.140  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.140  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.140  4203  4694 D BSLib   : 
08-03 15:10:35.140  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x
08-03 15:10:35.140  4203  4694 D BSLib   : 01 - Max 1024 fds
08-03 15:10:35.140  4203  4694 D BSLib   : 
08-03 15:10:35.140  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.140  4203  4694 D BSLib   : 
08-03 15:10:35.141  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.141  4203  4694 D BSLib   : 
08-03 15:10:35.141  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.141  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.141  4203  4694 D BSLib   : 
08-03 15:10:35.142  4203  4694 D BSLib   : main: fd_listen: fd=1204
08-03 15:10:35.142  4203  4694 D BSLib   :  flags=0x01 - Max 1024 fds
08-03 15:10:35.142  4203  4694 D BSLib   : 
08-03 15:10:35.142  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.142  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.142  4203  4694 D BSLib   : 
08-03 15:10:35.142  4203  4694 D BSLib   : 
08-03 15:10:35.143  4203  4694 D BSLib   : main: fd_listen: fd=1204
08-03 15:10:35.143  4203  4694 D BSLib   :  flags=0x01 - Max 1024 fds
08-03 15:10:35.143  4203  4694 D BSLib   : 
08-03 15:10:35.143  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.143  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.143  4203  4694 D BSLib   : 
08-03 15:10:35.144  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.144  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.144  4203  4694 D BSLib   : 
08-03 15:10:35.144  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.144  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.144  4203  4694 D BSLib   : 
08-03 15:10:35.144  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.144  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.144  4203  4694 D BSLib   : 
08-03 15:10:35.144  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.144  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.144  4203  4694 D BSLib   : 
08-03 15:10:35.145  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x
08-03 15:10:35.145  4203  4694 D BSLib   : 01 - Max 1024 fds
08-03 15:10:35.145  4203  4694 D BSLib   : 
08-03 15:10:35.145  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.145  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.145  4203  4694 D BSLib   : 
08-03 15:10:35.145  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.145  4203  4694 D BSLib   : 1204 flags=0x
08-03 15:10:35.145  4203  4694 D BSLib   : 01 - Max 
08-03 15:10:35.145  4203  4694 D BSLib   : 1024 fds
08-03 15:10:35.145  4203  4694 D BSLib   : 
08-03 15:10:35.148  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.148  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.148  4203  4694 D BSLib   : 
08-03 15:10:35.148  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.148  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.148  4203  4694 D BSLib   : 
08-03 15:10:35.148  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.148  4203  4694 D BSLib   : 
08-03 15:10:35.148  4203  4694 D BSLib   : main: fd_listen: fd=1204
08-03 15:10:35.148  4203  4694 D BSLib   :  flags=0x01 - Max 1024 fds
08-03 15:10:35.148  4203  4694 D BSLib   : 
08-03 15:10:35.149  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.149  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.149  4203  4694 D BSLib   : 
08-03 15:10:35.149  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.149  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.149  4203  4694 D BSLib   : 
08-03 15:10:35.149  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.149  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.149  4203  4694 D BSLib   : 
08-03 15:10:35.149  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.149  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.149  4203  4694 D BSLib   : 
08-03 15:10:35.150  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.150  4203  4694 D BSLib   : 
08-03 15:10:35.150  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.150  4203  4694 D BSLib   : 
08-03 15:10:35.150  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x0
08-03 15:10:35.150  4203  4694 D BSLib   : 1 - Max 1024 fds
08-03 15:10:35.150  4203  4694 D BSLib   : 
08-03 15:10:35.150  4203  4694 D BSLib   : main: fd_listen: fd=1204
08-03 15:10:35.150  4203  4694 D BSLib   :  flags=0x01 - Max 1024 fds
08-03 15:10:35.150  4203  4694 D BSLib   : 
08-03 15:10:35.151  4203  4694 D BSLib   : main: fd_listen: fd=1204
08-03 15:10:35.151  4203  4694 D BSLib   :  flags=0x01 - Max 1024 fds
08-03 15:10:35.151  4203  4694 D BSLib   : 
08-03 15:10:35.151  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.151  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.151  4203  4694 D BSLib   : 
08-03 15:10:35.151  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.151  4203  4694 D BSLib   : 1204 flags=0x01 - Max 1024 fds
08-03 15:10:35.151  4203  4694 D BSLib   : 
08-03 15:10:35.151  4203  4694 D BSLib   : main: fd_listen: fd=1204 flags=0x01
08-03 15:10:35.151  4203  4694 D BSLib   :  - Max 1024 fds
08-03 15:10:35.151  4203  4694 D BSLib   : 
08-03 15:10:35.152  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.155  4203  4694 D BSLib   : 1204
08-03 15:10:35.155  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.155  4203  4694 D BSLib   : 0
08-03 15:10:35.155  4203  4694 D BSLib   : 1
08-03 15:10:35.155  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.155  4203  4694 D BSLib   : 1024
08-03 15:10:35.156  4203  4694 D BSLib   :  fds
08-03 15:10:35.156  4203  4694 D BSLib   : 
08-03 15:10:35.156  4203  4694 D BSLib   : 
08-03 15:10:35.156  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.156  4203  4694 D BSLib   : 1204
08-03 15:10:35.157  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.157  4203  4694 D BSLib   : 0
08-03 15:10:35.157  4203  4694 D BSLib   : 1
08-03 15:10:35.157  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.157  4203  4694 D BSLib   : 1024
08-03 15:10:35.157  4203  4694 D BSLib   :  fds
08-03 15:10:35.157  4203  4694 D BSLib   : 
08-03 15:10:35.158  4203  4694 D BSLib   : 
08-03 15:10:35.158  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.158  4203  4694 D BSLib   : 1204
08-03 15:10:35.158  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.158  4203  4694 D BSLib   : 0
08-03 15:10:35.158  4203  4694 D BSLib   : 1
08-03 15:10:35.158  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.158  4203  4694 D BSLib   : 1024
08-03 15:10:35.158  4203  4694 D BSLib   :  fds
08-03 15:10:35.158  4203  4694 D BSLib   : 
08-03 15:10:35.158  4203  4694 D BSLib   : 
08-03 15:10:35.158  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.158  4203  4694 D BSLib   : 1204
08-03 15:10:35.159  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.159  4203  4694 D BSLib   : 0
08-03 15:10:35.159  4203  4694 D BSLib   : 1
08-03 15:10:35.159  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.159  4203  4694 D BSLib   : 1024
08-03 15:10:35.159  4203  4694 D BSLib   :  fds
08-03 15:10:35.159  4203  4694 D BSLib   : 
08-03 15:10:35.159  4203  4694 D BSLib   : 
08-03 15:10:35.159  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.159  4203  4694 D BSLib   : 1204
08-03 15:10:35.159  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.159  4203  4694 D BSLib   : 0
08-03 15:10:35.159  4203  4694 D BSLib   : 1
08-03 15:10:35.159  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.159  4203  4694 D BSLib   : 1024
08-03 15:10:35.159  4203  4694 D BSLib   :  fds
08-03 15:10:35.159  4203  4694 D BSLib   : 
08-03 15:10:35.160  4203  4694 D BSLib   : 
08-03 15:10:35.160  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.160  4203  4694 D BSLib   : 1204
08-03 15:10:35.160  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.160  4203  4694 D BSLib   : 0
08-03 15:10:35.160  4203  4694 D BSLib   : 1
08-03 15:10:35.160  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.160  4203  4694 D BSLib   : 1024
08-03 15:10:35.160  4203  4694 D BSLib   :  fds
08-03 15:10:35.160  4203  4694 D BSLib   : 
08-03 15:10:35.160  4203  4694 D BSLib   : 
08-03 15:10:35.160  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.160  4203  4694 D BSLib   : 1204
08-03 15:10:35.160  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.160  4203  4694 D BSLib   : 0
08-03 15:10:35.160  4203  4694 D BSLib   : 1
08-03 15:10:35.160  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.161  4203  4694 D BSLib   : 1024
08-03 15:10:35.161  4203  4694 D BSLib   :  fds
08-03 15:10:35.161  4203  4694 D BSLib   : 
08-03 15:10:35.161  4203  4694 D BSLib   : 
08-03 15:10:35.161  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.161  4203  4694 D BSLib   : 1204
08-03 15:10:35.161  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.161  4203  4694 D BSLib   : 0
08-03 15:10:35.161  4203  4694 D BSLib   : 1
08-03 15:10:35.161  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.161  4203  4694 D BSLib   : 1024
08-03 15:10:35.161  4203  4694 D BSLib   :  fds
08-03 15:10:35.161  4203  4694 D BSLib   : 
08-03 15:10:35.161  4203  4694 D BSLib   : 
08-03 15:10:35.161  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.161  4203  4694 D BSLib   : 1204
08-03 15:10:35.161  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.162  4203  4694 D BSLib   : 0
08-03 15:10:35.162  4203  4694 D BSLib   : 1
08-03 15:10:35.162  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.162  4203  4694 D BSLib   : 1024
08-03 15:10:35.162  4203  4694 D BSLib   :  fds
08-03 15:10:35.162  4203  4694 D BSLib   : 
08-03 15:10:35.162  4203  4694 D BSLib   : 
08-03 15:10:35.162  4203  4694 D BSLib   : main: fd_listen: fd=
08-03 15:10:35.162  4203  4694 D BSLib   : 1204
08-03 15:10:35.163  4203  4694 D BSLib   :  flags=0x
08-03 15:10:35.163  4203  4694 D BSLib   : 0
08-03 15:10:35.163  4203  4694 D BSLib   : 1
08-03 15:10:35.163  4203  4694 D BSLib   :  - Max 
08-03 15:10:35.163  4203  4694 D BSLib   : 1024
08-03 15:10:35.163  4203  4694 D BSLib   :  fds
08-03 15:10:35.163  4203  4694 D BSLib   : 
08-03 15:10:35.163  4203  4694 D BSLib   : stream: rtp_listen failed: af=AF_INET ports=1024-49152 (Too many open files)
08-03 15:10:35.164  4203  4694 D BSLib   : stream: failed to create socket for media 'audio' (Too many open files)
08-03 15:10:35.164  4203  4694 D BSLib   : audio: destroyed (started=0)
08-03 15:10:35.164  4203  4694 D BSLib   : ua: call_alloc: Too many open files
08-03 15:10:36.489  4203  4694 D BSLib   : ua: incoming OPTIONS message from sip:asterisk@192.168.20.200 (192.168.20.200:5060)
08-03 15:10:36.490  4203  4694 D BSLib   : ua: using best effort AF: af=AF_INET
08-03 15:10:36.490  4203  4694 D BSLib   : call: alloc with params laddr=192.168.20.131, af=AF_INET, use_rtp=0
08-03 15:10:36.490  4203  4694 D BSLib   : call: use_video=0
08-03 15:10:36.491  4203  4694 D BSLib   : audio: destroyed (started=0)
08-03 15:10:36.491  4203  4693 D BSLib   : ua event (REGISTER_OK) 200 OK
08-03 15:10:40.658  4203  4693 D BSLib   : ua event (CALL_INCOMING) sip:2348167728034@192.168.20.200
08-03 15:10:40.659  4203  4694 D BSLib   : ua: sipsess connect via UDP 192.168.20.200:5060 --> 192.168.20.131:49339
08-03 15:10:40.659  4203  4694 D BSLib   : ua: using AF from sdp offer: af=AF_INET
08-03 15:10:40.659  4203  4694 D BSLib   : call: alloc wi
08-03 15:10:40.659  4203  4694 D BSLib   : th params laddr=192.168.20.131, af=AF_INET, use_rtp=1
08-03 15:10:40.659  4203  4694 D BSLib   : call: use_video=0

**Comment by sreimers** [comment, marked as answer]
At first you should check if connections are closed (with `fd_close`) properly. if so you can try to increase `maxfds` with:

```diff
diff --git a/src/main.c b/src/main.c
index 31060a78..6274fb12 100644
--- a/src/main.c
+++ b/src/main.c
@@ -107,10 +107,14 @@ int main(int argc, char *argv[])
 
        err = libre_init();
        if (err)
                goto out;
 
+       err = fd_setsize(2048);
+       if (err)
+               goto out;
+
        tmr_init(&tmr_quit);
 
 #ifdef HAVE_GETOPT
        for (;;) {
                const int c = getopt(argc, argv, "46a:de:f:p:hu:n:vst:m:");
```

**Comment by assegaf** [comment]
> At first you should check if connections are closed (with `fd_close`) properly. if so you can try to increase `maxfds` with:
> 

I see so its about network connection, not about file open, I will find out more which causing this based on "fd_close", meanwhile I will set to maximum 20 times of 1024, I think will sufficient.

**Reply by assegaf** [reply]
is there any way/command in android environment (root access available), to see if connection (sip call) is hanging after its said suucessfully hang up ?
